compiler/src/iree/compiler/Codegen/Common/GPU/GPUMaterializeEncoding.cpp

```diff
@@ -168,6 +148,30 @@ static void interleave(MaterializeEncodingInfo::Swizzle &swizzle, int srcIndex,
   swizzle.permutation = outPermutation;
 }
 
+// Returns the index of the dimension whose flattened size (flattening inner
+// dimensions into it) matches the given `targetSize`. This is used to compute
+// interleaving indices.
+//
+// Example:
+//    Input shape = [16, 8, 4, 4]
+//    Input targetSize = 16
+// -> Return 2, because the tail of the shape starting at index 2 is [4, 4],
+//    whose product equals targetSize.
```

Comment:
Is this function doing what the comment says? `getDimIdxForTargetSize` initializes `size` to `targetSize`, so I think it is actually returning the index if the dimension whose flattened size matches `targetSize / shape[0]`. I think it is better to initialize `size` to `1`, and pass `targetPreferredLoadBitWidth / ABits / swizzle.expandShape[srcIdx][0]` as `targetSize` below.

Reply:
Haha, you made this change as I was reviewing. Looks good now!